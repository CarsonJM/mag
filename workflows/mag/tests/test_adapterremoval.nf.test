nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_adapterremoval"

    test("Parameters: clip_tool = 'adapterremoval'") {

        when {
            params {
                input                       = 'https://raw.githubusercontent.com/nf-core/test-datasets/mag/samplesheets/samplesheet.euk.csv'
                clip_tool                   = 'adapterremoval'
                keep_phix                   = true
                centrifuge_db               = null
                kraken2_db                  = null
                skip_krona                  = true
                skip_megahit                = true
                skip_spades                 = true
                skip_spadeshybrid           = true
                skip_quast                  = true
                skip_prodigal               = true
                skip_binning                = true
                skip_binqc                  = true
                skip_gtdbtk                 = true
                skip_prokka                 = true
                skip_metaeuk                = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out,
                ).match()}
            )
        }
    }
}
