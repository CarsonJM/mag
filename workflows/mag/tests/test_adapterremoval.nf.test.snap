{
    "Parameters: clip_tool = 'adapterremoval'": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_minigut_euk",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_euk_run0_ar2.pair1.truncated.fastq.gz:md5,e8f01a4fdf2e348330fc02a47eb3c5af",
                            "test_minigut_euk_run0_ar2.pair2.truncated.fastq.gz:md5,06510940505288d80a3d940db85d0ed5"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2_euk",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_sample2_euk_run0_ar2.pair1.truncated.fastq.gz:md5,92b8643e591f7490c980fe4d0879c955",
                            "test_minigut_sample2_euk_run0_ar2.pair2.truncated.fastq.gz:md5,7d4436e9532bb9665080b77efe441faf"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,eca9f4c6e135848b1247916040e52b99"
                ],
                "assemblies": [
                    
                ],
                "bin_summary": [
                    
                ],
                "genomad": [
                    
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    
                ],
                "refined_unbins": [
                    
                ],
                "short_reads": [
                    [
                        {
                            "id": "test_minigut_euk",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_euk_run0_ar2.pair1.truncated.fastq.gz:md5,e8f01a4fdf2e348330fc02a47eb3c5af",
                            "test_minigut_euk_run0_ar2.pair2.truncated.fastq.gz:md5,06510940505288d80a3d940db85d0ed5"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2_euk",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_sample2_euk_run0_ar2.pair1.truncated.fastq.gz:md5,92b8643e591f7490c980fe4d0879c955",
                            "test_minigut_sample2_euk_run0_ar2.pair2.truncated.fastq.gz:md5,7d4436e9532bb9665080b77efe441faf"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,eca9f4c6e135848b1247916040e52b99"
                ]
            }
        ],
        "timestamp": "2024-02-05T23:39:32.916744986"
    }
}